File: OtherLanguages/PHP/CrashReporter/Data/CrashInstance.php
<?
class CrashInstance
{
	public $id;
	public $crashId;
	public $time;

	public function read($reader)
	{
		$this->id = $reader->hasField(0) ? $reader->readInteger(0) : null;
		$this->crashId = $reader->readInteger(1);
		$this->time = $reader->readStructure(2);
	}

	public function write($writer)
	{
		if ($this->id!=null) $writer->writeValue(0, $this->id);
		$writer->writeValue(1, $this->crashId);
		$writer->writeStructure(2, $this->time);
	}
}
?>

File: Sitana.Framework/CrashReporter/Data/CrashInstance.cs
using System;
using Ebatianos.DataTransfer;

namespace Ebatianos.CrashReporter.Data
{
	public class CrashInstance: IEcsStructure
	{
		public int? Id;
		public int CrashId;
		public DateAndTime Time;

        void IEcsStructure.Read(EcsReader reader)
		{
			Id = reader.HasField(0) ? (int?)reader.ReadInt32(0) : null;
			CrashId = reader.ReadInt32(1);
			Time = reader.ReadStructure<DateAndTime>(2);
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			if (Id!=null) writer.Write(0, Id.Value);
			writer.Write(1, CrashId);
			writer.Write(2, Time);
		}
	}
}